src/Bicep.Cli/Arguments/FormatArguments.cs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.FileSystem;
+using Bicep.Core.PrettyPrint.Options;
+using System;
+using System.IO;
+
+namespace Bicep.Cli.Arguments
+{
+    public class FormatArguments : ArgumentsBase
+    {
+        public FormatArguments(string[] args) : base(Constants.Command.Format)
+        {
+            for (var i = 0; i < args.Length; i++)
+            {
+                switch (args[i].ToLowerInvariant())
+                {
+                    case "--stdout":
+                        OutputToStdOut = true;
+                        break;
+
+                    case "--outdir":
+                        if (args.Length == i + 1)
+                        {
+                            throw new CommandLineException($"The --outdir parameter expects an argument");
+                        }
+                        if (OutputDir is not null)
+                        {
+                            throw new CommandLineException($"The --outdir parameter cannot be specified twice");
+                        }
+                        OutputDir = args[i + 1];
+                        i++;
+                        break;
+
+                    case "--outfile":
+                        if (args.Length == i + 1)
+                        {
+                            throw new CommandLineException($"The --outfile parameter expects an argument");
+                        }
+                        if (OutputFile is not null)
+                        {
+                            throw new CommandLineException($"The --outfile parameter cannot be specified twice");
+                        }
+                        OutputFile = args[i + 1];
+                        i++;
+                        break;
+
+                    case "--newline":
+                        if (args.Length == i + 1)
+                        {
+                            throw new CommandLineException($"The --newline parameter expects an argument");
+                        }
+                        if (Newline is not null)
+                        {
+                            throw new CommandLineException($"The --newline parameter cannot be specified twice");
+                        }
+                        try
+                        {
+                            Newline = (NewlineOption)Enum.Parse(typeof(NewlineOption), value: args[i + 1], true);
```

Comment:
Most `Parse` methods have a `TryParse` equivalent which allows you to avoid the exception handling - e.g.

```csharp
if (!Enum.TryParse<NewlineOption>(args[i + 1], true, out Newline))
{
    throw new CommandLineException($"The --newline parameter only accepts values: {string.Join(" | ", Enum.GetNames(typeof(NewlineOption)))}");
}
```

Reply:
Done!